### PR TITLE
Publishing zones as marker array

### LINKED ISSUE #328: Multiple Zonesets may be shown at the same time
### Commit
e00a5ab

### Steps to reproduce
Switch back and forth between the two zones quickly

### Expected behavior
Only one set of (warning and safety) zones is shown at any time.

### Observed behavior
Screenshot from 2022-06-22 10-37-06

### Test case
https://github.com/PilzDE/psen_scan_v2/blob/08905198eef5ed2cd8d2803cdfda90aff0a7bc40/test/acceptance_tests/acceptancetest_multiple_zones_display.md

### PR DESCRIPTION
**Pull request by ct2034**
fixes #328 
<details>
<summary>PR Checklist</summary>

### Things to add, update or check by the maintainers before this PR can be merged.

- [x] Public api function documentation
- [x] Architecture documentation reflects actual code structure
- [x] [Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)
- [x] Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)
- [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
- [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [x] CHANGELOG.rst updated
- [x] Copyright headers
- [x] Examples

### Review Checklist
- [x] Soft- and hardware architecture (diagrams and description)
- [x] Test review (test plan and individual test cases)
- [x] Documentation describes purpose of file(s) and responsibilities
- [x] Code (coding rules, style guide)

### Release planning (please answer)
- [x] When is the new feature released? => ASAP
- [x] ~Which dependent packages do have to be released simultaneously?~

### Hardware tests
_Unstrike the text below to enable automatic hardware tests if available. Otherwise use this as a request for the reviewer._
- [x] ~Perform hardware tests~

</details>

### CODE REVIEW

**Comment by ct2034** [comment]
do we still need this file:
https://github.com/PilzDE/psen_scan_v2/blob/eed0925f10712b477fb87e99c5be7da1a7f987ed/scripts/zonesets_visualization.py

File: src/active_zoneset_node.cpp [hunk 1]
```diff
@@ -93,35 +91,33 @@ void ActiveZonesetNode::addMarkers(std::vector<visualization_msgs::Marker>& new_
...
   for (const auto& marker : new_markers)
   {
-    zoneset_marker_.publish(marker);
+    current_markers_.push_back(marker);
   }
-  last_markers_ = std::move(new_markers);
 }
 
-void ActiveZonesetNode::deleteLastMarkers()
+void ActiveZonesetNode::addDeleteForUnusedLastMarkers()
```
Comment (rfeistenauer) [inline src/active_zoneset_node.cpp:98]:
```suggestion
void ActiveZonesetNode::addDeleteMessageForUnusedLastMarkers()
```

File: test/acceptance_tests/acceptancetest_multiple_zones_display.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Acceptance test to verify the bug where multiple zones are displayed at the same time. 
+
+[#328](https://github.com/PilzDE/psen_scan_v2/issues/328)
```
Comment (rfeistenauer) [inline test/acceptance_tests/acceptancetest_multiple_zones_display.md:3]:
```suggestion
# Acceptance test to verify that multiple zones are not displayed simultaniously. 

Definition of the identified error case can be found in : [Issue #328](https://github.com/PilzDE/psen_scan_v2/issues/328)
```

File: test/acceptance_tests/acceptancetest_multiple_zones_display.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Setup
+Make sure your scanner has a configuration with two different zone sets, for example like [this file](acceptancetest_multiple_zones_display.xml)
+
+Then make sure, the launchfile `acceptancetest_multiple_zones_display.launch` also uses that config.
+
+## Expected output
```
Comment (rfeistenauer) [inline test/acceptance_tests/acceptancetest_multiple_zones_display.md:10]:
```suggestion
## Test procedure
```

File: test/acceptance_tests/acceptancetest_multiple_zones_display.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    r10 = rospy.Rate(10)  # 10hz
+    while not rospy.is_shutdown():
+        pub.publish(0)
+        r10.sleep()
+        pub.publish(8)
+        r10.sleep()
+        pub.publish(0)
+        if random.random() > .5:
+            r10.sleep()
+            pub.publish(8)
```
Comment (rfeistenauer) [inline test/acceptance_tests/acceptancetest_multiple_zones_display.py:22]: would make it easier to understand what is happening if you would extract the relay_cmd byte and sleeps.

```suggestion
        switch_to_zone(zone_id=0)
        switch_to_zone(zone_id=1)
        switch_to_zone(zone_id=0)
        if random.random() > .5:
            switch_to_zone(zone_id=1)
```

**Review by rfeistenauer — COMMENTED** [review]
Just a few small things

File: test/acceptance_tests/acceptancetest_multiple_zones_display.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+if __name__ == "__main__":
+    rospy.init_node("multiple_zones_test")
+
+    pub = rospy.Publisher("/relay_cmd", Byte, queue_size=10)
+
+    def publish_zone_and_sleep(zone: int):
+        if zone == 0:
+            pub.publish(0)
+        elif zone == 1:
+            pub.publish(1)
```
Comment (rfeistenauer) [inline test/acceptance_tests/acceptancetest_multiple_zones_display.py:16]: Shouldn't the second publish be a '8'?

Comment (ct2034) [reply test/acceptance_tests/acceptancetest_multiple_zones_display.py:16]: thanks :)

**Pull request merged**